proteuslib/tools/parameter_sweep.py

```diff
@@ -15,18 +15,92 @@
 import sys
 import os
 import itertools
+import warnings
 
+from enum import IntEnum, auto
+from abc import abstractmethod, ABC 
 from idaes.core.util import get_solver
 
 # ================================================================
 
+class SamplingType(IntEnum):
```

Comment:
Just out of curiosity, can we use `Enum` here or is `IntEnum` necessary here? If `IntEnum` _is_ necessary, is it because `auto()` is being used?

Reply:
This is a good catch -- I looked it up and `IntEnum` will silently cast its members to `int` for comparison purposes. E.g.
```python
class FooType(Enum):
    BAR = 1

class BarType(IntEnum):
    FOO = 1

# False
FooType.BAR == 1

# True
BarType.FOO == 1
```

The `auto()` function is just for convenience.